Automattic/wp-calypso
# Apply Jetpack cloud base styles to landing page

### Changes proposed in this Pull Request

Jetpack cloud has some base styles that are imported in the sidebar component. The landing page doesn't have a sidebar and therefore these styles aren't applied there. This PR fixes this by importing styles conditionally in the main layout component.

It is a follow up of #51139, which has been reverted because it inadvertently modified styles in Calypso blue.

This update makes base components have the same styles across the platform, which will make fixing contrast issues easier in a next PR.

### Implementation notes

#51139 moved the loading of the styles from the sidebar to the masterbar (which is used in the landing page), but that was just moving the problem. This PR goes a step further by conditionally loading base styles in cloud only, from the main layout component. Doing a conditional loading client side is not an issue in our case since none of cloud sections are rendered server side.

This PR touches the main layout only, which means that some pages like the login page or the 404 page won't load cloud base styles at the moment. If this implementation is accepted, we can consider porting it to those sections as well.

### Testing instructions

Download the PR and run both Calypso and cloud

In Calypso

- Browse some sections in Calypso and check that they look like production (specifically, visit `/reader` since the first PR introduced styling issues in that page)

In Jetpack cloud

- Visit the landing page
- Check that the 3 states mentioned below match production (to easily simulate these states, edit the `noSite` middleware locally at `client/jetpack-cloud/controller.ts`). Note that the input placeholder and secondary button match the rest of the site rather than the landing page in production.
- Browse some other sections in cloud and check that they look like production

### Screenshots

_No site_
<img width="817" alt="111485540-131e2780-870d-11eb-8081-cbc35b75c28e" src="https://user-images.githubusercontent.com/1620183/112050186-edd35400-8b26-11eb-9505-245c70164bbe.png">


_No visible site_
![111487332-a572fb00-870e-11eb-834b-b89171389a45](https://user-images.githubusercontent.com/1620183/112050171-eb70fa00-8b26-11eb-9170-7a127dfcb9c9.png)


_1+ site_
![111487302-9e4bed00-870e-11eb-805b-6e8930c662a4](https://user-images.githubusercontent.com/1620183/112050156-e8760980-8b26-11eb-8ab8-a98e42a678e9.png)
